agent/reconciler/host_reconciler.go

```diff
@@ -171,6 +212,30 @@ func (r *HostReconciler) getBootstrapScript(ctx context.Context, dataSecretName,
 	return bootstrapSecret, nil
 }
 
+func (r *HostReconciler) executeScript(ctx context.Context, script string) error {
+	cmd := exec.CommandContext(ctx, "/bin/bash", "-c", script)
+	cmd.Stderr = os.Stderr
+	cmd.Stdout = os.Stdout
+	if err := cmd.Run(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *HostReconciler) parseScript(ctx context.Context, script string) (string, error) {
+	parser, err := template.New("parser").Parse(script)
```

Comment:
Can you please reuse the existing utility:
https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/blob/b09a31d78eb0b2e824f3f98dad96f2ead0f638a8/agent/cloudinit/template_parser.go#L21-L36

Reply:
@mayur-tolexo I also tried this template parser but was not working

Reply:
not working? Almost the same thing is done in `ParseTemplate()` method as that you are doing in `parseScript()` function.  
Is it throwing an error or after parsing, "BundleDownloadPath" value is not getting substituted?

Reply:
@khannakshat7 also use this opportunity to fix anything in the existing helper if needed :)

Reply:
@anusha94 This is fixed. I'm using the existing template parser and it is working fine